File: apps/admin/src/types/reservation.ts
export type ReservationStatus = 'pending' | 'confirmed' | 'in_progress' | 'completed' | 'cancelled' | 'no_show';

export interface Reservation {
  id: string;
  client_id: string;
  client_resource_id: string;
  service_id: string;
  assigned_to: string | null;
  scheduled_at: string;
  estimated_end: string;
  status: ReservationStatus;
  notes: string | null;
  cancelled_at: string | null;
  cancel_reason: string | null;
  created_by: string;
  created_at: string;
  client_resource?: { label: string | null; data: Record<string, unknown> | null; plate: string | null; brand: string | null; model: string | null; color: string | null };
  service?: { name: string; price: string };
  client?: { name: string; email: string };
}

export interface AvailableSlot {
  start: string;
  end: string;
  available: number;
}

File: apps/staff/lib/features/reservations/domain/entities/reservation.dart
import '../enums/reservation_status.dart';

class Reservation {
  final String id;
  final String clientResourceId;
  final String serviceId;
  final String? assignedTo;
  final DateTime scheduledAt;
  final DateTime estimatedEnd;
  final ReservationStatus status;
  final String? notes;
  final String? vehiclePlate;
  final String? vehicleBrand;
  final String? serviceName;
  final String? servicePrice;
  final String? clientName;

  const Reservation({
    required this.id,
    required this.clientResourceId,
    required this.serviceId,
    this.assignedTo,
    required this.scheduledAt,
    required this.estimatedEnd,
    required this.status,
    this.notes,
    this.vehiclePlate,
    this.vehicleBrand,
    this.serviceName,
    this.servicePrice,
    this.clientName,
  });
}
